File: 01_Workshop10LeetCode/C#/Algorithms/TwoSun/TwoSunAgain.cs
public class TwoSumAgain
{
    public int[] TwoSumDictionaryAgain(int[] values, int target)
    {
        var map = new Dictionary<int, int>();
        for (int i = 0; i < values.Length; i++)
        {
            var currentValue = values[i];
            if (map.ContainsKey(currentValue))
            {
                return new int[] { map[currentValue], i };
            }

            var complement = target - currentValue;
            map[complement] = i;
        }
        throw new Exception("No such value");
    }
}

File: 01_Workshop10LeetCode/TS/src/TwoSun/TwoSunAgain.ts
export class TwoSunAgain{
    public TwoSumDictionaryAgain(nums: number[], target: number): number[]{
        const map = new Map<number, number>();
        for(let i =0; i < nums.length; i++){
            const currentValue = nums[i];
            if(map.has(currentValue)){
                return [map.get(currentValue)!, i];
            }
            let complement = target - nums[i];
            map.set(complement, i);
        }
        throw new Error("Not found");
    }
}